File: docs/剑指offer/Python/07-斐波那契数列.py
# -*- coding:utf-8 -*-
class Solution:
    def Fibonacci(self, n):
        # write code here
        tempArray = [0,1]
        if n >= 2:
            for i in range(2, n+1):
                tempArray[i%2] = tempArray[0] + tempArray[1]
        return tempArray[n%2]

File: docs/jianzhioffer/javascript/07-斐波那契数列.js
function Fibonacci(n)
{
    // write code here
    if (n===0 || n===1) {
        return n;
    }
    let n1 = 0, n2=1, sum=1;
    for(let i=2;i<=n;i++) {
        sum = n1+n2;
        n1=n2;
        n2=sum;
    }
    return sum;
}
module.exports = {
    Fibonacci : Fibonacci
};
